### ISSUE TITLE
Passing an unpacked tuple of types as type arguments to the type parameters `[T, *Ts]` gets error

### ISSUE THREAD

**Issue by hyperkai** [microsoft/pyright#11276]
*Memo:
- pyright test.py
- pyright 1.1.408
- Python 3.14.0

Passing an unpacked tuple of types as type arguments to the type parameters `[T, *Ts]` gets the error as shown below:

*Memo:
- mypy and ty don't get error.

```python
from typing import Unpack

class Cls[T, *Ts]: ...

v1: Cls[*tuple[int, str, int, str]]        # Error
v2: Cls[Unpack[tuple[int, str, int, str]]] # Error
```

> error: Unpacked arguments cannot be used in this context

**Comment by erictraut** [comment]
That's currently by design. An unpacked argument must align to a TypeVarTuple in the argument list. 

While it would be possible to support this in specific cases where the `*tuple[]` does not include elements with unknown length, it would require a lot of extra complexity in the type checker. Since this isn't a common use case, it is not supported.

I'll change this from a bug report to an enhancement request. I'm going to mark it as closed because I don't have any plans to implement it at this time. I might reconsider in the future if there's sufficient demand.
